### ISSUE TITLE
Disabling libs `blocking-elements` & `wicg-inert` for performance

### ISSUE THREAD

**Issue by IgorNovozhilov** [material-components/material-web#2745]
When using `mwc-dialog`, I ran into the problem that libraries `blocking-elements` & `wicg-inert` perform modifications of all DOM elements except for the contents of the dialog.

Given that the site is quite large, the delay in rendering turns out to be very long.
In my case, it takes about 100 ms to update every 2000 DOM elements.

Updated attributes: `aria-hidden`, `inert`, `tabindex`

**Describe the solution you'd like**
Is it possible to make an optional connection of these libraries? [mwc-dialog-base.ts#L10 --> import 'blocking-elements';](https://github.com/material-components/material-web/blob/master/packages/dialog/mwc-dialog-base.ts#L10)
> For me, at the moment, performance plays a more important role than the correctness of the specified attributes

**Describe alternatives you've considered**
At the moment, I have "dirty" silenced the blocking of elements, since the `class BlockingElementsImpl` is global (`document.$blockingElements = new BlockingElementsImpl()` [blocking-elements.ts#L436](https://github.com/PolymerLabs/blocking-elements/blob/master/src/blocking-elements.ts#L436))
```js
  ...
  Object.defineProperty(document, '$blockingElements', {
    set: () => { },
    get: () => ({ push: () => { }, remove: () => { } })
  })
  await import('https://unpkg.com/@material/mwc-dialog?module')
  ...
```

**Comment by asyncLiz** [comment]
Can you give an example of your HTML structure? `blocking-elements` only sets `inert` on siblings up to the path of the top blocking element. Having 2000 nodes as siblings along the path seems unusual and could be mitigated by restructuring your DOM.

```html
<!-- These are siblings, each one will be set to inert -->
<!-- If there are thousands of siblings, this will be a problem -->
<button>Button Sibling</button>
<button>Button Sibling</button>
<mwc-dialog></mwc-dialog>
```

```html
<!-- Alternatively, restructuring the DOM can reduce the number of siblings -->
<!-- Only the one div sibling will be set to inert -->
<div class="sibling">
  <button>Button child</button>
  <button>Button child</button>
</div>
<mwc-dialog></mwc-dialog>
```
```

**Comment by IgorNovozhilov** [comment]
@asyncLiz

> Can you give an example of your HTML structure?

no, it's a big commercial ~legacy~ :crying_cat_face:  project

> only sets inert on siblings up to the path of the top blocking element

this condition is valid only for `inert`. I was talking about all the attributes (aria-hidden, inert, tabindex) in the aggregate
the main pain is brought by changing the tabindex in https://unpkg.com/wicg-inert@3.1.1/dist/inert.esm.js?module

The smallest page contains this:
```
document.querySelectorAll('[tabindex]').length
> 473
```

**Comment by IgorNovozhilov** [comment]
And Chrome DevTools Console begins to 'gently' remind me that something is wrong :turtle:
```
﻿https://unpkg.com/@material/dialog@13.0.0-canary.65125b3a6.0/foundation.js?module
foundation.js?module: 150 [Violation] 'setTimeout' handler took 101ms
```

**Comment by asyncLiz** [comment]
Right, but setting `inert` (which sets the attributes) will only set the attributes for sibling elements. It sounds like the HTML structure could be optimized from where the `<mwc-dialog>` is and the number of siblings it has along the path to the topmost blocking element.

**Comment by yinonov** [comment]
Just put the `mwc-dialog` as top as possible in your DOM tree and set its content by some [portal practice](https://material.angular.io/cdk/portal)
